### ISSUE TITLE
Weird behaviour with assertion errors

### ISSUE THREAD

**Issue by AndreaCrotti** [funcool/promesa#114]
I encountered this behaviour which I found a bit confusing.
So it looks like throwables behave as expected, but if I have for example an AssertionError being thrown then it just hangs forever (in this example I just killed it manually).
Is that expected? I suppose it's because Throwables are handled but not other exceptions, but is that on purpose?
Thanks

```clojure
p.core> @(create (fn [resolve reject]
                   (throw (ex-info "" {}))))
Execution error (ExceptionInfo) at promesa.core/eval10540$fn (REPL:40).

p.core> @(create (fn [resolve reject]
                  (assert (not= 1 2))))
Execution error (InterruptedException) at java.util.concurrent.CompletableFuture/reportGet (CompletableFuture.java:386).
null
```

**Comment by niwinz** [comment]
This is because we catching `Exception` instead of `Throwable`. Will be fixed in next version, thanks.

**Comment by niwinz** [comment]
Should be fixed in 9.0.466, thanks for reporting the issue.
